services/vision/mlvision/ml_model.go

```diff
@@ -232,3 +219,81 @@ func checkClassificationScores(in []float64) []float64 {
 	}
 	return in // no need to sigmoid
 }
+
+// Number interface for converting between numbers.
+type number interface {
+	constraints.Integer | constraints.Float
+}
+
+// convertNumberSlice converts any number slice into another number slice.
+func convertNumberSlice[T1, T2 number](t1 []T1) []T2 {
+	t2 := make([]T2, len(t1))
+	for i := range t1 {
+		t2[i] = T2(t1[i])
+	}
+	return t2
+}
+
+func convertToFloat64Slice(slice interface{}) ([]float64, error) {
```

Comment:
this is such a silly function but it was very useful since tensors really care about the data type that you feed them.